mobsf/MobSF/views/authorization.py

```diff
@@ -0,0 +1,169 @@
+            sig = signature(view)
+            arguments = sig.bind(request, *args, **kwargs)
+            api = arguments.arguments.get('api')
+            if settings.DISABLE_AUTHENTICATION == '1' or api:
+                # Disable authorization for all functions
+                return view(request, *args, **kwargs)
+            # Enforce authorization for all web function calls
+            return pr(
+                perm,
+                raise_exception=True)(view)(request, *args, **kwargs)
+        return wrapper
+    return decorator
+
+
+def has_permission(request, permission, api):
+    """Check if user has permission."""
+    if request.user.is_staff:
+        return True
+    if settings.DISABLE_AUTHENTICATION == '1' or api:
+        return True
+    if not request.user.has_perm(permission):
+        return False
+    return True
+
+
+def create_authorization_roles():
+    """Create Authorization Roles."""
+    maintainer, _created = Group.objects.get_or_create(name=MAINTAINER_GROUP)
+    Group.objects.get_or_create(name=VIEWER_GROUP)
+
+    scan_permissions = Permission.objects.filter(
+        codename=PERM_CAN_SCAN)
+    suppress_permissions = Permission.objects.filter(
+        codename=PERM_CAN_SUPPRESS)
+    delete_permissions = Permission.objects.filter(
+        codename=PERM_CAN_DELETE)
+    all_perms = list(chain(
+        scan_permissions, suppress_permissions, delete_permissions))
+    maintainer.permissions.set(all_perms)
+
+
+@login_required
+@staff_member_required
+def users(request):
+    """Show all users."""
+    users = get_user_model().objects.all()
+    context = {
+        'title': 'All Users',
+        'users': users,
+        'version': settings.MOBSF_VER,
+    }
+    return render(request, 'auth/users.html', context)
+
+
+@login_required
+@staff_member_required
+def create_user(request):
+    if request.method == 'POST':
+        form = RegisterForm(request.POST)
+        if form.is_valid():
+            role = request.POST.get('role')
+            username = request.POST.get('username')
+            if not username:
+                messages.error(request, 'No Username Provided')
+                return redirect('create_user')
+            if not USERNAME_REGEX.match(username):
+                messages.error(request, 'Invalid Username')
+                return redirect('create_user')
+            user = form.save()
+            user.is_staff = False
+            if role == 'maintainer':
+                user.groups.add(Group.objects.get(name=MAINTAINER_GROUP))
+            else:
+                user.groups.add(Group.objects.get(name=VIEWER_GROUP))
+            messages.success(
+                request,
+                'User created successfully!')
+            return redirect('create_user')
+        else:
+            messages.error(
+                request,
+                'Please correct the error below.')
+            return redirect('create_user')
+    else:
+        form = RegisterForm()
```

Comment:
Do we want to set any requirements for the password complexity? it will come as request soon I guess...

Reply:
Good point, I will add a password validator.